src/Field/ScaleCombo/ScaleCombo.jsx

```diff
@@ -193,29 +200,41 @@ class ScaleCombo extends React.Component {
    * based on an existing instance of {@link Ol.Map}
    */
   getOptionsFromMap = () => {
+    const {
+      map,
+      resolutionsFilter
+    } = this.props;
+
     if (!isEmpty(this.state.scales)) {
       Logger.debug('Array with scales found. Returning');
       return;
     }
-    if (!this.props.map) {
+    if (!map) {
       Logger.warn('Map component not found. Could not initialize options array.');
       return;
     }
 
-    let map = this.props.map;
     let mv = map.getView();
     // use existing resolutions array if exists
     let resolutions = mv.getResolutions();
     if (isEmpty(resolutions)) {
       for (let currentZoomLevel = mv.getMaxZoom(); currentZoomLevel >= mv.getMinZoom(); currentZoomLevel--) {
         let resolution = mv.getResolutionForZoom(currentZoomLevel);
-        this.pushScale(resolution, mv);
+        if (isFunction(resolutionsFilter)) {
```

Comment:
You may want to define a default resolutionsFilter like `() => true` to omit this check here (and all over).